app/models/assignment.rb

```diff
@@ -0,0 +1,6 @@
+class Assignment < ApplicationRecord
+  belongs_to :user
+  belongs_to :offspring
+  belongs_to :shift
+  validates :user, presence: true
```

Comment:
You should validate shift, but the rest can be nil.
Perhaps offspring is also needed, as when you nullify it you should destroy the assignment, but the user is optional (i.e. you can change it as an administrator)